### PR TITLE
Issue #2010: NTH_VALUE

### PR DESCRIPTION
**Pull request by hawkfish**
Implement and test.

### CODE REVIEW

File: src/execution/operator/aggregate/physical_window.cpp [hunk 1]
```diff
@@ -761,6 +761,21 @@ static void ComputeWindowExpression(BoundWindowExpression *wexpr, ChunkCollectio
 		case ExpressionType::WINDOW_LAST_VALUE:
 			payload_collection.CopyCell(0, bounds.window_end - 1, result, output_offset);
 			break;
+		case ExpressionType::WINDOW_NTH_VALUE: {
+			if (payload_collection.ColumnCount() != 2) {
+				throw BinderException("NTH_VALUE needs a parameter");
```
Comment (Mytherin) [inline src/execution/operator/aggregate/physical_window.cpp:766]: Could you add a test that triggers this?

File: src/parser/transform/expression/transform_function.cpp [hunk 2]
```diff
@@ -154,6 +156,13 @@ unique_ptr<ParsedExpression> Transformer::TransformFuncCall(duckdb_libpgquery::P
 					if (function_list.size() > 3) {
 						throw ParserException("Incorrect number of parameters for function %s", lowercase_name);
 					}
+				} else if (win_fun_type == ExpressionType::WINDOW_NTH_VALUE) {
+					if (function_list.size() > 1) {
+						expr->children.push_back(move(function_list[1]));
+					}
+					if (function_list.size() > 2) {
+						throw ParserException("Incorrect number of parameters for function %s", lowercase_name);
```
Comment (Mytherin) [inline src/parser/transform/expression/transform_function.cpp:164]: Could you add a test that triggers this?

File: test/sql/window/test_basic_window.test [hunk 3]
```diff
@@ -73,6 +73,48 @@ SELECT empno, first_value(empno) OVER (PARTITION BY depname ORDER BY empno) fv F
...
+personnel	2	5
+personnel	5	5
+sales	1	4
+sales	3	4
+sales	4	4
+
+# nth_value
+query III
+SELECT depname, empno,
+	nth_value(empno, 2) OVER (
```
Comment (Mytherin) [inline test/sql/window/test_basic_window.test:100]: Could you add tests:
* Where either of the the parameters is a constant NULL
* Where either of the parameters is a column that contains NULL values
* Where the second parameter (offset) is not a constant
* Where the first parameter is a constant

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great. Some comments:

File: test/sql/window/test_basic_window.test (hunk 3 above)
Comment (hawkfish) [reply test/sql/window/test_basic_window.test:116]: At some point these should be done for all window functions. Breaking up `test_basic_window` seems like the way to go.

**Comment by Mytherin** [comment]
Thanks, looks great now!

**Pull request merged**